clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -586,4 +588,59 @@ public void patch(
       }
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof KubernetesV2Credentials)) {
+      return false;
+    }
+
+    KubernetesV2Credentials that = (KubernetesV2Credentials) o;
+    return accountName.equals(that.accountName)
+        && Objects.equals(namespaces, that.namespaces)
+        && Objects.equals(omitNamespaces, that.omitNamespaces)
+        && Objects.equals(kinds, that.kinds)
+        && Objects.equals(omitKinds, that.omitKinds)
+        && Objects.equals(customResources, that.customResources)
+        && Objects.equals(kubectlExecutable, that.kubectlExecutable)
+        && Objects.equals(kubectlRequestTimeoutSeconds, that.kubectlRequestTimeoutSeconds)
+        && Objects.equals(kubeconfigFile, that.kubeconfigFile)
+        && Objects.equals(kubeconfigFileHash, that.kubeconfigFileHash)
+        && Objects.equals(serviceAccount, that.serviceAccount)
+        && Objects.equals(context, that.context)
+        && Objects.equals(onlySpinnakerManaged, that.onlySpinnakerManaged)
+        && Objects.equals(liveManifestCalls, that.liveManifestCalls)
+        && Objects.equals(checkPermissionsOnStartup, that.checkPermissionsOnStartup)
+        && Objects.equals(cachingPolicies, that.cachingPolicies)
+        && Objects.equals(oAuthServiceAccount, that.oAuthServiceAccount)
+        && Objects.equals(oAuthScopes, that.oAuthScopes)
+        && Objects.equals(metrics, that.metrics)
```

Comment:
`metrics` is not `final` and can be changed (from `true` to `false` if the metrics aren't readable).  This means that the current value may not be the configured one, similar to with `omitKinds`.